packages/workspace/src/workspace/nacl_files/multi_env/multi_env_source.ts

```diff
@@ -316,12 +318,28 @@ const buildMultiEnvSource = (
     return buildRes.changes
   }
 
+  const getElementsSource = async (env?: string): Promise<ElementsSource> => (
+    (await getState()).states[env ?? primarySourceName].elements
+  )
+
   const getElementIdsBySelectors = async (
     selectors: ElementSelector[],
-    commonOnly = false,
+    fromSource = 'env',
     compact = false,
   ): Promise<AsyncIterable<ElemID>> => {
-    const relevantSource = commonOnly ? commonSource() : primarySource()
+    let relevantSource: ElementsSource
+    switch (fromSource) {
+      case 'env': {
+        relevantSource = primarySource()
+        break
+      }
+      case 'common': {
+        relevantSource = commonSource()
+        break
+      }
+      default: relevantSource = await getElementsSource()
+    }
```

Comment:
nit: you can turn this into a function and then `relevantSource` can remain `const` (and you can also avoid the `breaks`)
```suggestion
    const determineSource = async (): Promise<ElementsSource> => {
      switch (fromSource) {
        case 'env': {
          return primarySource()
        }
        case 'common': {
          return commonSource()
        }
        default: {
          return getElementsSource()
        }
      }
    }
    const relevantSource = await determineSource()
```